.github/workflows/main.yml

```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-latest, ubuntu-20.04, ubuntu-22.04]
+        os: [macos-13, macos-14, ubuntu-20.04, ubuntu-22.04]
```

Comment:
I think that leaving the latest macOS version will be enough since the differences between the last two releases are usually minimized by Developer tools package updates and brew package updates. The only reason why this could matter is the choice of the default compiler, but I think we should rather steer the default choice in CMake rather than test the default on all possible platforms

Reply:
There are two versions of macOS because I tested macOS on arm64. Runner-image `macos-13` runs on arm64 device. In such case it had better to test macos-14 and macos-14-xlarge because we have special logic in cmake for arm64
![image](https://github.com/ostis-ai/sc-machine/assets/56268020/f053c221-0ec8-467c-8333-369157f466f2)

Reply:
Now runner-image for ubuntu-24.04 is available in test mode. We can start using it. I think it helps us to watch out users problems on ubuntu-24.04. And I have install ubuntu-24.04 and now I use it during development.
![image](https://github.com/ostis-ai/sc-machine/assets/56268020/c62f74c8-08fd-4076-8f4c-ba28b6aa8900)
This PR fixes all warning and problems for ubuntu-24.04 as well as for the latest macos.

As for support for testing on older versions of ubuntu, I'd leave them. Now sc-machine doesn't have anything that doesn't work on ubuntu-24.04 and works on ubuntu-20.04 and vica versa. I think we can continue to support testing sc-machine on ubuntu-20.04. There is no need to delete this support.